### PR TITLE
Fix/positions v3

### PR DESCRIPTION
**Pull request by vuonghuuhung**
## Summary by CodeRabbit

- **New Features**
  - Enhanced pool detail component with accurate liquidity and volume fetching.
  - Introduced a new hook for managing liquidity volume data across components.
  - Improved handling of price data in position-related components for better accuracy.

- **Bug Fixes**
  - Enhanced error handling for fetching operations in various components.
  - Fixed potential issues with displaying total volume values.

- **Refactor**
  - Streamlined data fetching logic and state management in multiple components to improve performance and clarity.
  - Updated component structures to ensure accurate rendering based on new data flows.

### CODE REVIEW

File: src/pages/Pool-V3/components/PositionList/index.tsx [hunk 1]
```diff
@@ -52,7 +52,7 @@ const PositionList = () => {
     })();
 
     return () => {};
-  }, [address, poolList, positions, isLight, isFetchingPositions]);
+  }, [address, poolList.length, positions, isLight, isFetchingPositions, poolPrice.length]);
```
Comment (quangdz1704) [inline src/pages/Pool-V3/components/PositionList/index.tsx]: same above => poolPrice.length

Comment (quangdz1704) [reply src/pages/Pool-V3/components/PositionList/index.tsx]: poolList is state or variable, if poolList is state, you shouldn't use .length

File: src/pages/Pool-V3/hooks/useGetPoolLiquidityVolume.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      refetchOnWindowFocus: true,
+      placeholderData: [],
+      cacheTime: 5 * 60 * 1000
+    }
+  );
+
+  useEffect(() => {
+    if (data.length === 0 || prices.length === 0) return;
+    data.forEach((item) => {
+      //   if (item.poolDayData.nodes.length > 0) {
```
Comment (quangdz1704) [inline src/pages/Pool-V3/hooks/useGetPoolLiquidityVolume.ts:24]: remove unused comment

File: src/rest/graphClient.ts [hunk 3]
```diff
@@ -170,6 +170,64 @@ export const getPoolsLiquidityAndVolume = async (): Promise<PoolLiquidityAndVolu
...
+  poolDayData: {
+    nodes: {
+      volumeTokenX: number;
+      volumeTokenY: number;
+    }[];
+  };
+};
+
+export const getPoolsLiqudityAndVolumeAmount = async (): Promise<PoolLiquidityAndVolumeAmount[]> => {
+  const yesterdayIndex = Math.floor(Date.now() / (24 * 60 * 60 * 1000)) - 1;
```
Comment (quangdz1704) [inline src/rest/graphClient.ts:195]: alter magic number to constant

File: src/pages/Pool-V3/components/PoolDetail/index.tsx [hunk 4]
```diff
@@ -111,10 +112,10 @@ const PoolV3Detail = () => {
       });
     };
 
-    if (poolDetail && prices && liquidity && poolDetail.poolKey === poolKeyString) {
+    if (poolDetail && poolPrice && liquidity && poolDetail.poolKey === poolKeyString) {
       getAPRInfo();
     }
-  }, [liquidity, feeDailyData, poolDetail, prices, poolKeyString]);
+  }, [liquidity.total, feeDailyData, poolDetail, poolPrice.length, poolKeyString]);
```
[the remark this answers is not in the document]
Comment (vuonghuuhung) [reply src/pages/Pool-V3/components/PoolDetail/index.tsx]: you made my day

**Review by quangdz1704 — APPROVED** [review]

**Review by quangdz1704 — APPROVED** [review]

**Review by haunv3 — APPROVED** [review]

**Pull request merged**